src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,222 @@
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        int editedLines;
+        String query = "UPDATE cars SET is_deleted = TRUE "
+                + "WHERE is_deleted = FALSE AND id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement = connection.prepareStatement(query)) {
+            updateCarStatement.setLong(1, id);
+            editedLines = updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot delete car from DB by id: " + id, e);
+        }
+        if (editedLines > 0) {
+            deleteDrivers(id);
+        }
+        return editedLines > 0;
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT cars.id AS id, model, manufacturer_id, "
+                + "manufacturers.name AS manufacturer_name, "
+                + "manufacturers.country AS manufacturer_country "
+                + "FROM cars "
+                + "JOIN manufacturers ON cars.manufacturer_id = manufacturers.id "
+                + "JOIN cars_drivers ON cars.id = cars_drivers.car_id "
+                + "WHERE cars.is_deleted = FALSE AND cars_drivers.driver_id = ?";
+        List<Car> carList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsByDriverStatement = connection
+                        .prepareStatement(query)) {
+            getAllCarsByDriverStatement.setLong(1, driverId);
+            ResultSet generatedKeys = getAllCarsByDriverStatement.executeQuery();
+            while (generatedKeys.next()) {
+                carList.add(parseCarWithManufacturerFromResultSet(generatedKeys));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get all cars from DB by driver id: "
+                    + driverId, e);
+        }
+        for (Car car : carList) {
+            car.setDrivers(getDriversForCar(car));
+        }
+        return carList;
+    }
+
+    private void insertDrivers(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement addDriversToCarStatement = connection
+                        .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            addDriversToCarStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                addDriversToCarStatement.setLong(2, driver.getId());
+                addDriversToCarStatement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot insert to cars_drivers drivers: "
+                    + car.getDrivers(), e);
+        }
+    }
+
+    private void deleteDrivers(Long id) {

```

Comment:
In this project we use soft delete, so there is no need in this method

Reply:
if we want to update drivers, we need to delete all previous data about drivers and rewrite it. that's from theory video.

Reply:
if we want to update car's drivers, we delete them, and then we rewrite them. that's from the theory video.